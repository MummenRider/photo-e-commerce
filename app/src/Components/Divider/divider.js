import React from "react";
import {
  Frame,
  Inner,
  Title,
  Subtitle,
  ImageContent,
  Button,
  ButtonText,
  ButtonIcon,
  ButtonIconPath,
} from "./divider.style";

export default function Divider({ children, ...restProps }) {
  return (
    <Frame {...restProps}>
      <Inner>{children}</Inner>
    </Frame>
  );
}

Divider.Title = function DividerTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Divider.Subtitle = function DividerSubtitle({ children, ...restProps }) {
  return <Subtitle {...restProps}>{children}</Subtitle>;
};

Divider.ImageContent = function DividerImageContent({ ...restProps }) {
  return <ImageContent {...restProps} />;
};

Divider.Button = function DividerButton({
  destination,
  children,
  ...restProps
}) {
  return (
    <Button onClick={destination} {...restProps}>
      {children}
    </Button>
  );
};

Divider.ButtonText = function DividerButtonText({ children, ...restProps }) {
  return <ButtonText {...restProps}>{children}</ButtonText>;
};

Divider.ButtonIcon = function DividerButtonIcon({ children, ...restProps }) {
  return <ButtonIcon {...restProps}>{children}</ButtonIcon>;
};
Divider.ButtonIconPath = function DividerButtonIconPath({ ...restProps }) {
  return <ButtonIconPath {...restProps} />;
};
