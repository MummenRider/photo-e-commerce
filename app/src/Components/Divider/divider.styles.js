import { motion } from "framer-motion";
import styled from "styled-components";

export const Frame = styled.section`
  width: 100%;
  background-color: ${(props) => props.bkgColor};
  height: ${(props) => props.height};
  min-height: ${(props) => props.minHeight};
  color: ${(props) => props.color};
  position: relative;
`;
export const Inner = styled(motion.div)`
  width: 90%;
  text-align: center;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  height: 70%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
`;

export const Title = styled.h1`
  font-size: clamp(3.5rem, ${(props) => props.size}, 20rem);
  font-family: broadacre-thin-1, sans-serif;
  font-weight: 100;
  font-style: normal;
  font-weight: lighter;
  line-height: 1.5;
  letter-spacing: 3px;
  text-transform: uppercase;
`;

export const Subtitle = styled.h3`
  font-family: var(--ff-primary);
  font-size: clamp(1.2rem, 0.8vw, 15rem);
  letter-spacing: 4px;
  font-weight: 400;
  padding: 2rem 0;
`;

export const TextFrame = styled.div``;
export const SVGFrame = styled(motion.div)`
  path {
    fill: black;
  }
`;

export const ImageFrame = styled(motion.img)`
  height: 20vh;
  width: auto;

  @media screen and (min-width: 1030px) {
    height: 30vh;
  }
`;
