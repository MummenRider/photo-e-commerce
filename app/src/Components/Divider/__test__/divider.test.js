import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import Divider from "Components/Divider/divider";
import flower from "Assets/Images/flower.png";

afterEach(cleanup);

describe("<Divider />", () => {
  it("should render <Divider /> with button", () => {
    const handleClick = jest.fn();
    const { container, getByText } = render(
      <Divider>
        <Divider.Title>Some title here</Divider.Title>
        <Divider.Button destination={handleClick}>
          <Divider.ButtonText>Click me</Divider.ButtonText>
          <Divider.ButtonIcon>
            <Divider.ButtonIconPath />
          </Divider.ButtonIcon>
        </Divider.Button>
      </Divider>
    );
    expect(handleClick).toHaveBeenCalledTimes(0);
    fireEvent.click(getByText("Click me"));
    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render <Divider /> with data included", () => {
    const { container, getByText, getByAltText } = render(
      <Divider>
        <Divider.Title>Some title here</Divider.Title>
        <Divider.Subtitle>
          This is subtitle included in divider component
        </Divider.Subtitle>
        <Divider.ImageContent src={flower} alt="divider image" />
      </Divider>
    );
    expect(getByText("Some title here")).toBeTruthy();
    expect(
      getByText("This is subtitle included in divider component")
    ).toBeTruthy();
    expect(getByAltText("divider image")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render <Divider /> with just title", () => {
    const { container, getByText, queryByText } = render(
      <Divider>
        <Divider.Title>Some title here</Divider.Title>
      </Divider>
    );
    expect(getByText("Some title here")).toBeTruthy();
    expect(
      queryByText("This is subtitle included in divider component")
    ).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
