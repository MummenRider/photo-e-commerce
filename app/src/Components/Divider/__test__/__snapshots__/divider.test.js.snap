// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> should render <Divider /> with button 1`] = `
<section
  class="sc-bdnxRM iCphrG"
>
  <div
    class="sc-gtsrHT qrQIS"
  >
    <h1
      class="sc-dlnjwi hZvaQd"
    >
      Some title here
    </h1>
    <button
      class="sc-jSFjdj gvvKvs"
    >
      <span
        class="sc-gKAaRy rxnOu"
      >
        Click me
      </span>
      <svg
        class="sc-iCoGMd jeAZbZ"
      >
        <path
          class="sc-fujyAs bLaEBj"
        />
      </svg>
    </button>
  </div>
</section>
`;

exports[`<Divider /> should render <Divider /> with data included 1`] = `
<section
  class="sc-bdnxRM iCphrG"
>
  <div
    class="sc-gtsrHT qrQIS"
  >
    <h1
      class="sc-dlnjwi hZvaQd"
    >
      Some title here
    </h1>
    <h1
      class="sc-hKFxyN"
    >
      This is subtitle included in divider component
    </h1>
    <img
      alt="divider image"
      class="sc-eCApnc"
      src="flower.png"
    />
  </div>
</section>
`;

exports[`<Divider /> should render <Divider /> with just title 1`] = `
<section
  class="sc-bdnxRM iCphrG"
>
  <div
    class="sc-gtsrHT qrQIS"
  >
    <h1
      class="sc-dlnjwi hZvaQd"
    >
      Some title here
    </h1>
  </div>
</section>
`;
